pydicer/convert/data.py

```diff
@@ -198,21 +211,75 @@ def convert(self):
                     )
 
                     write_nrrd_from_mask_directory(output_dir, nrrd_file)
+
+                    # Save JSON
+                    json_file_name = f"{series_uid_hash}_{linked_uid_hash}.nrrd"
+                    json_file = self.output_directory.joinpath(
+                        patient_id,
+                        study_id_hash,
+                        "structures",
+                        f"{series_uid_hash}_{linked_uid_hash}.json",
+                    )
+                    convert_dicom_headers(
+                        rt_struct_file.file_path, f"{series_uid_hash}_{linked_uid_hash}", json_file
+                    )
+
                 elif sop_class_uid == PET_IMAGE_STORAGE_UID:
 
-                    # all_files = file_dic["files"]
                     series_files = df_files.file_path.tolist()
                     series_files = [str(f) for f in series_files]
 
-                    output_file = self.output_directory.joinpath(
-                        patient_id, study_id_hash, "images", f"PT_{series_uid_hash}.nii.gz"
+                    output_file_base = f"PT_{series_uid_hash}"
+                    nifti_file_name = f"{output_file_base}.nii.gz"
+                    nifti_file = self.output_directory.joinpath(
+                        patient_id, study_id_hash, "images", nifti_file_name
                     )
-                    output_file.parent.mkdir(exist_ok=True, parents=True)
+                    nifti_file.parent.mkdir(exist_ok=True, parents=True)
 
                     convert_dicom_to_nifti_pt(
                         series_files,
-                        output_file,
+                        nifti_file,
+                    )
+
+                    json_file_name = f"{output_file_base}.json"
+                    json_file = self.output_directory.joinpath(
+                        patient_id, study_id_hash, "images", json_file_name
+                    )
+                    convert_dicom_headers(series_files[0], nifti_file_name, json_file)
+
+                elif sop_class_uid == RT_PLAN_STORAGE_UID:
```

Comment:
Great working on adding the RT Plan too, the output directories are starting to look nice and structured when we run the pipeline